src/Nethermind/Nethermind.Init/Steps/InitializePrecompiles.cs

```diff
@@ -5,34 +5,36 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Nethermind.Api;
+using Nethermind.Core.Specs;
 using Nethermind.Crypto;
 using Nethermind.Logging;
 
 namespace Nethermind.Init.Steps;
 
 public class InitializePrecompiles : IStep
 {
-    private readonly INethermindApi _api;
+    private ILogger _logger;
+    private IInitConfig _initConfig;
+    private ISpecProvider _specProvider;
 
-    public InitializePrecompiles(INethermindApi api)
+    public InitializePrecompiles(ISpecProvider specProvider, IInitConfig initConfig, ILogManager logManager)
     {
-        _api = api;
+        _specProvider = specProvider;
+        _initConfig = initConfig;
+        _logger = logManager.GetClassLogger<InitializePrecompiles>();
```

Comment:
Can we inject ILogger's directly instead of LogManager's?

Reply:
Probably can make work.

Reply:
Best I can do is `public InitializePrecompiles(ISpecProvider specProvider, IInitConfig initConfig, ILogger<InitializePrecompiles> logger)` need to have the type somewhere. Or need to pass a name.

Reply:
Can you get the type from the type you are injecting to?

Reply:
I think it could be done with custom middleware?

Reply:
Found it.